File: algorithm/leetcode5448.path-crossing.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        vector<vector<int>> dir = {{0,1}, {1,0},{-1,0},{0,-1}};
        set<pair<int,int>> s;
        s.insert({0,0});
        int x = 0, y = 0;
        for(int i = 0; i < path.size(); i++)
        {
            if(path[i] == 'N')
                y++;
            else if(path[i] == 'S')
                y--;
            else if(path[i] == 'E')
                x++;
            else
                x--;
            if(s.find({x,y}) != s.end())
                return true;
            else
                s.insert({x,y});
        }
        return false;
    }   
};

File: algorithm/Python3_version/leetcode5448.path-crossing.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        s = set()
        s.add(tuple([0,0]))
        x = 0
        y = 0
        for i in range(len(path)):
            if path[i] == 'N':
                y += 1
            elif path[i] == 'S':
                y -= 1
            elif path[i] == 'E':
                x += 1
            else:
                x -= 1
            if tuple([x,y]) in s:
                return True;
            else:
                s.add(tuple([x,y]))
        return False